README.md

```diff
@@ -60,22 +76,53 @@ val value by intSort
 val expr = (array.select(index - 1.expr) lt value) and
         (array.select(index + 1.expr) gt value)
 ```
-Check out our [example project](examples) for more complicated examples.
 
-### Solver agnostic SMT formula representation
-KSMT provides a solver-independent formula representation
-with back and forth transformations to the solver's native representation.
-Such representation allows introspection of formulas and transformation of formulas 
-independent of the solver.
+With this DSL, you can
+* transform expressions from the solver native representation to KSMT representation and vice versa,
+* use _formula introspection_,
+* manipulate expressions without involving a solver,
+* use expressions even if the solver is freed.
+
+Expression interning (hash consing) affords faster expression comparison: we do not need to compare the expression
+trees. Expressions are deduplicated, so we avoid redundant memory allocation.
+
+### Utilities to simplify and transform expressions
+
+KSMT provides a simplification engine applicable to all supported expressions for all supported theories in a
+solver-agnostic model:
```

Comment:
It is better to omit `in a solver-agnostic model`. The model is a representation of the SMT solution. We can apply simplification to any expressions regardless of whether it is in the model or not. The only connection between the simplification engine and the model is that the simplification engine is used for expression evaluation (i.e. simplification wrt the model).